src/base-client-helpers_test.ts

```diff
@@ -0,0 +1,141 @@
+import { assertEquals } from "https://deno.land/std@0.185.0/testing/asserts.ts";
+import { _internals, getUserAgent } from "./base-client-helpers.ts";
+import { serializeData } from "./base-client-helpers.ts";
+import { assertSpyCalls, stub } from "./dev_deps.ts";
+
+Deno.test(_internals.getModuleVersion.name, async (t) => {
+  await t.step(
+    "should return the version if the module is sourced from deno.land",
+    () => {
+      const getModuleUrlStub = stub(_internals, "getModuleUrl", () => {
+        return new URL("https://deno.land/x/deno_slack_api@2.1.0/mod.ts)");
+      });
+
+      try {
+        const moduleVersion = _internals.getModuleVersion();
+
+        assertSpyCalls(getModuleUrlStub, 1);
+        assertEquals(moduleVersion, "2.1.0");
+      } finally {
```

Comment:
For my own education, why do we wrap the assertions in try/finally? What would happen here if an exception is raised in `getModuleVersion()`?

Reply:
From what I can understand the resources get subbed on a global level and must be explicitly released :shrug:  an error is raised if another test tries to stub a stubbed resource

The `try/finally` pattern insures that the resource gets released with `getModuleUrlStub.restore();`, I've also thought of using [bdd](https://deno.land/manual@v1.36.3/basics/testing/behavior_driven_development) but opted for this pattern since it followed existing testing patterns